pingdom/pingdom.py

```diff
@@ -0,0 +1,77 @@
+import sys
+import requests
+import json
+import argparse
+
+parser = argparse.ArgumentParser(description='Collects monitoring data from Pingdom.')
+parser.add_argument('-u', '--pingdom-user-name', help='The Pingdom User Name', required=True)
+parser.add_argument('-p', '--pingdom-password', help='The Pingdom Password', required=True)
+parser.add_argument('-a', '--pingdom-api-key', help='The Pingdom API-KEY', required=True)
+
+class Pingdom:
+    def __init__(self, api_key, user_name, password):
+        self.api_key = api_key,
+        self.user_name = user_name,
+        self.password = password
+
+    def handle_error(self, error_message):
+        sys.stderr.write("ERROR:|Pingdom| " + error_message)
+        sys.exit(1)
+
+    def call_api(self, api):
+        headers = {'App-Key': self.api_key[0]}
+
+        base_api = 'https://api.pingdom.com/api/2.0/' + api
+        response = requests.get(base_api, headers=headers, auth=requests.auth.HTTPBasicAuth(self.user_name[0], self.password))
+
+        if response.status_code == 200:
+            return response.json()
+
+        else:
+            self.handle_error("API [" + base_api + "] failed to execute with error code [" + str(response.status_code) + "].")
+            return None
+
+    def get_checks(self):
+        response = self.call_api('checks')
+
+        if None != response:
+            data = response.get("checks")
+            counts = response.get("counts")
+            up_count = 0
+            down_count = 0
+            unconfirmed_down_count = 0
+            unknown_count = 0
+            paused_count = 0
+
+            for x in data:
+                status = x.get("status")
+                if status == "up":
+                    up_count = up_count + 1
+                elif status == "down":
+                    down_count == down_count + 1
+                elif status == "unconfirmed_down":
+                    unconfirmed_down_count = unconfirmed_down_count + 1
+                elif status == "unknown":
+                    unknown_count = unknown_count + 1
+                elif status == "paused":
+                    paused_count = paused_count + 1
+
+            counts["up_count"] = up_count
+            counts["down_count"] = down_count
+            counts["unconfirmed_down"] = unconfirmed_down_count
+            counts["unknown"] = unknown_count
+            counts["paused"] = paused_count
+
+            data.append(counts)
+
+            print(json.dumps(data))
+        else:
+            print(json.dumps([]))
```

Comment:
Ditto. Looks like you may not want to exit from handle_error or at least make it an optional flag and exit only under certain conditions.